hepcrawl/spiders/phil_spider.py

```diff
@@ -0,0 +1,168 @@
+       a HEPRecord (or a request to scrape for the pdf file if link exists).
+
+
+    Example usage:
+    scrapy crawl phil -s "JSON_OUTPUT_DIR=tmp/"
+    scrapy crawl phil -a source_file=file://`pwd`/tests/responses/phil/inspire_15-12-2015.json -s "JSON_OUTPUT_DIR=tmp/"
+
+
+    TODO:
+    *Have to check if new records are appended to the file or
+     if the file is just replaced with new information:
+     Actually some old records are removed while new ones are added?
+
+
+    Happy crawling!
+    """
+
+    name = 'phil'
+    start_urls = ["http://philpapers.org/philpapers/raw/export/inspire.json"]
+
+    def __init__(self, source_file=None, *args, **kwargs):
+        """Construct Phil spider."""
+        super(PhilSpider, self).__init__(*args, **kwargs)
+        self.source_file = source_file
+
+    def start_requests(self):
+        """You can also run the spider on local test files."""
+        if self.source_file:
+            yield Request(self.source_file)
+        elif self.start_urls:
+            for url in self.start_urls:
+                yield Request(url)
+
+    def get_authors(self, author_element):
+        """Parses the line where there are data about the author(s)."""
+        authors = []
+        for auth in author_element:
+            split_author = split_fullname(auth)
+            surname = split_author[0]
+            given_names = split_author[-1]
+            authors.append({
+                # Automatic fullname creation behaves strangely here
+                'full_name': surname + ", " + given_names,
+                'surname': surname,
+                'given_names': given_names,
+            })
+
+        return authors
+
+    def get_date(self, record):
+        """Return a standard format date.
+
+        YYYY-MM-DD, YYYY-MM or YYYY.
+        """
+        date_raw = record['year'].split("/")
+        if len(date_raw) == 1:
+            date_published = date_raw[0]
+        elif len(date_raw) == 2:
+            date_published = date_raw[-1] + "-" + date_raw[0]
+        elif len(date_raw) == 3:
+            date_published = date_raw[-1] + "-" + date_raw[1] + "-" + date_raw[0]
+
+        return date_published
+
+    def parse(self, response):
+        """Parse Philpapers JSON file into a HEP record."""
+
+        jsonresponse = json.loads(response.body_as_unicode())
+        for jsonrecord in jsonresponse:
+            urls_in_record = jsonrecord.get("links")
+            if urls_in_record:
+                link = urls_in_record[0]
+                request = Request(link, callback=self.scrape_for_pdf)
+                request.meta["urls"] = urls_in_record
+                request.meta["jsonrecord"] = jsonrecord
+                yield request
+            else:
+                response.meta["urls"] = []
+                request.meta["jsonrecord"] = jsonrecord
+                yield self.build_item(response)
+
+    def scrape_for_pdf(self, response):
```

Comment:
Test coverage decreased somewhat. Perhaps you can try adding a test of the scraping of PDFs. See e.g. suggestions for the DNB spider about mocking the request using `responses`.